lib/topic_assigner.rb

```diff
@@ -189,7 +189,7 @@ def assign(assign_to, silent: false)
       )
     end
 
-    if SiteSetting.assign_mailer_enabled
+    if SiteSetting.assign_mailer == 'always' || (SiteSetting.assign_mailer == 'different_users' && @assigned_by.id != assign_to.id)
```

Comment:
I've come to hate magic strings. Can we instead have an enum that is used everywhere? (like we do for `Post.types` for example)